frontend/src/cljs/imas_seamap/map/subs.cljs

```diff
@@ -65,6 +65,13 @@
 (defn map-base-layers [{:keys [map]} _]
   (select-keys map [:grouped-base-layers :active-base-layer]))
 
+(defn categories-map [db _]
+  (let [categories (get-in db [:map :categories])]
+    (reduce
+     (fn [categories {:keys [name] :as category}]
```

Comment:
Second time in this PR I've seen this pattern (I always zoom in on uses of reduce, it's a prime candidate for "actually there's an existing library fn that does this").  How about extracting a util to index a collection by a key?  (If there isn't one already).  Have a bit of a play; one option is just to extract the categories case, another might be to pull out the map-key function, eg (into-map :name categories) or something like that (not claiming that's the best name)

Reply:
I've extracted the function, as after some searching I've found examples of similar problems, which are not solved by a library function (e.g. https://stackoverflow.com/questions/47082635/map-from-list-of-maps).